src/main/java/dataProvider/ConfigFileReader.java

```diff
@@ -0,0 +1,91 @@
+
+import enums.DriverType;
+
+import java.io.BufferedReader;
+import java.io.FileNotFoundException;
+import java.io.FileReader;
+import java.io.IOException;
+import java.util.Properties;
+
+public class ConfigFileReader {
+
+    private Properties properties;
+    private final String propertyFilePath = "src/test/resources/configs/configurations.properties";
+
+    public ConfigFileReader() {
+        BufferedReader bufferedReader;
+
+        try {
+            bufferedReader = new BufferedReader(new FileReader(propertyFilePath));
+            properties = new Properties();
+            try {
+                properties.load(bufferedReader);
+                bufferedReader.close();
+            } catch (IOException e) {
+                e.printStackTrace();
+            }
+        } catch (FileNotFoundException e) {
+            throw new RuntimeException("configurations.properties not found at " + propertyFilePath);
+        }
+    }
+
+    public String getDriverPath() {
+        String driverPath = properties.getProperty("driverPath");
+        if (driverPath != null) {
+            return driverPath;
+        } else {
+            throw new RuntimeException("driverPath not specified in the configurations.properties file" + propertyFilePath);
+        }
+    }
+
+    public long getImplicitWait() {
+        String implicitWait = properties.getProperty("implicitWait");
+        if (implicitWait != null) {
+            return Long.parseLong(implicitWait);
+        } else {
+            throw new RuntimeException("implicitWait not specified in the configurations.properties file" + propertyFilePath);
+        }
+    }
+
+    public String getWebsiteURL() {
+        String url = properties.getProperty("url");
+        if (url != null) {
+            return url;
+        } else {
+            throw new RuntimeException("url not specified in the configurations.properties file" + propertyFilePath);
+        }
+    }
+
+    public DriverType getBrowser() {
+        String browserName = properties.getProperty("browser");
+        if (browserName == null || browserName.equals("chrome")) {
+            return DriverType.CHROME;
+        } else if (browserName.equalsIgnoreCase("firefox")) {
+            return DriverType.FIREFOX;
+        } else if (browserName.equals("iexplorer")) {
+            return DriverType.INTERNET_EXPLORER;
+        } else {
+            throw new RuntimeException("Browser Name Key value in Configuration.properties is not matched : " + browserName);
+        }
+    }
+
+    public String getReportConfigPath() {
+        String reportConfigPath = properties.getProperty("reportConfigPath");
+        if (reportConfigPath != null) {
+            return reportConfigPath;
+        } else {
+            throw new RuntimeException("Report Config Path not specified in the configuration.propreties");
+        }
+    }
+
+    public String getLogConfigPath() {
+        String logConfigPath = properties.getProperty("logConfigPath");
```

Comment:
you have a lot of duplicated code, lines:
 - 33-40
 - 42-49
 - 51-58
 - 73-80 
 - 82-90
Make your code reusable, extract common code into a private method

Reply:
Done